frontend/src/component/feature/FeatureStrategy/FeatureStrategySegment/RecentlyUsedSegments/useRecentlyUsedSegments.ts

```diff
@@ -0,0 +1,28 @@
+import { useLocalStorageState } from 'hooks/useLocalStorageState';
+
+export const useRecentlyUsedSegments = (initialItems: number[] = []) => {
+    const [items, setItems] = useLocalStorageState<number[]>(
+        'recently-used-segments',
+        initialItems,
+    );
+
+    const addItem = (newItem: number | number[]) => {
+        setItems((prevItems) => {
+            const itemsToAdd = Array.isArray(newItem) ? newItem : [newItem];
+
+            let updatedItems = [...prevItems];
+            itemsToAdd.forEach((id) => {
+                updatedItems = updatedItems.filter(
+                    (existingId) => existingId !== id,
+                );
+                updatedItems = [id, ...updatedItems];
```

Comment:
Out of curiosity, I wanted to see if there was a way to insert at the start of an array, and there is! 
You could use [`unshift`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift):

```suggestion
                updatedItems.unshift(id);
```

Not sure it has any other benefits here, though. (and if there was a perf difference, this list is always gonna be so small that it don't matter anyway 🤷🏼 )

Reply:
It is generally good practice to not mutate array. Specially in react world. Due to how states etc work.

Reply:
Fair point. But in this case, you're creating a new array, anyway, so it doesn't really matter for states, right? It's not gonna be an issue here, but I have run into cases where spreading an object absolutely murdered the responsiveness because it had to chug through too many items. In that case, directly mutating the object instead was wildly more effective. That may or may not apply to arrays, but again: it was more out of curiosity 💁🏼